engine-tests/src/test/java/org/terasology/engine/persistence/serializers/TypeSerializerTest.java

```diff
@@ -150,7 +150,8 @@ public String toString() {
         }
     }
 
-    private static final class Animal<T> {
+    @SuppressWarnings("checkstyle:FinalClass")
```

Comment:
why does checkstyle think this should be final? I mean, it literally is extended a couple of lines later... is it because of the `private` constructor (which should probably be `protected` instead)?